app/components/rails_bootstrap/badge_component.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class RailsBootstrap::BadgeComponent < ApplicationComponent
+  # rename variant to variation 
+  # variant is a keyword in rails
+  VARIATIONS = ["primary", "secondary", "success", "danger", "warning", "light", "info", "dark"].freeze
+
+  validates :variation, inclusion: { in: VARIATIONS }
+  validates :pill, inclusion: { in: [true, false] }
+
+  def initialize(variation: "primary", pill: false)
+    @variation = variation
+    @pill = pill
+  end
+
+  def styles
+    "badge #{pill_class} #{variation_class}"
```

Comment:
The reason why I did not suggest you this way is because if pill_class returns `""` then it will cause unnecessary `"  "` (s spaces) in between `badges` and `badge-#{variation}`

Reply:
fixed it.